public/app/plugins/panel/graph2/GraphLegendEditor.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import capitalize from 'lodash/capitalize';
+import without from 'lodash/without';
+import { StatID, LegendOptions, PanelOptionsGroup, Switch } from '@grafana/ui';
+
+interface GraphLegendEditorProps {
+  stats: StatID[];
+  options: LegendOptions;
+  onChange: (options: LegendOptions) => void;
+}
+
+export const GraphLegendEditor: React.FunctionComponent<GraphLegendEditorProps> = props => {
+  const { stats, options, onChange } = props;
+
+  const onStatToggle = (stat: StatID) => (event?: React.SyntheticEvent<HTMLInputElement>) => {
+    let newStats;
+    if (!event) {
+      return;
+    }
+
+    // @ts-ignore
+    if (event.target.checked) {
+      newStats = (options.stats || []).concat([stat]);
+    } else {
+      newStats = without(options.stats, stat);
+    }
+    onChange({
+      ...options,
+      stats: newStats,
+    });
+  };
+
+  const onOptionToggle = (option: keyof LegendOptions) => (event?: React.SyntheticEvent<HTMLInputElement>) => {
+    const newOption = {};
+    if (!event) {
+      return;
+    }
+    // @ts-ignore
+    newOption[option] = event.target.checked;
+    onChange({
+      ...options,
+      ...newOption,
+    });
+  };
+
+  return (
+    <PanelOptionsGroup title="Legend">
+      <>
+        <h4>Legend options</h4>
+        <Switch label="Show legend" checked={options.isVisible} onChange={onOptionToggle('isVisible')} />
+        <Switch label="Display as table" checked={options.asTable} onChange={onOptionToggle('asTable')} />
+      </>
+
+      <>
+        <h4>Values</h4>
+        {stats.map(stat => {
```

Comment:
Another option would be using the StatsPicker:
```
<StatsPicker
        allowMultiple={true}
        stats={...}
```
The CSS needs work, but this will show the description and allow the list to grow longer without being too much in the UI

Reply:
Good point. This part of this code can be considered as 'legacy' as I have started working on it couple of weeks ago and haven't touched it since then. Gonna explore that anyways since it will make the UI more concise 👍